### ISSUE TITLE
[v5]Not able to run Multiremote with wdio test runner

### ISSUE THREAD

**Issue by artyompvl** [webdriverio/webdriverio#3257]
[//]: # NOTE: This repository only maintains packages that are listed in the [Readme](https://github.com/webdriverio/webdriverio/blob/master/README.md#packages). Please make sure that your issue is directly caused by one of these packages and if not file an issue in the correct 3rd party package repository.

**Environment (please complete the following information):**
 - **WebdriverIO version:** [5.2.0]
 - **Mode:** [WDIO Testrunner]
 - **If WDIO Testrunner, running sync/async:** [sync]
 - **Node.js version:** [8.11.4]
 - **NPM version:** [6.5.0]
 - **Browser name and version:** [Chrome  71]
 - **Platform name and version:** [Mac]
 - **Additional wdio packages used (if applicable):** [@wdio/selenium-standalone service]

**Config of WebdriverIO**
```
exports.config = {
    runner: 'local',
    //
    specs: [
        './qa/selenium-tests/soak/videoSoakLive.spec.js'
    ],
    //
    exclude: [
        // 'path/to/excluded/files'
    ],
    //
    maxInstances: 2,
    //
    capabilities: {
        maxInstances: 2,
        myChromeBrowser: {
            desiredCapabilities: {
                browserName: 'chrome'
            }
        },
        myFirefoxBrowser: {
            desiredCapabilities: {
                browserName: 'firefox'
            }
        }
    },
    //
    logLevel: 'trace',
    //
    deprecationWarnings: true,
    //
    bail: 0,
    //
    baseUrl: 'google.com/',
    //
    waitforTimeout: 10000,
    //
    connectionRetryTimeout: 90000,
    //
    connectionRetryCount: 3,
    //
    services: ['selenium-standalone'],
    //
    framework: 'mocha',
    //
    reporters: ['dot'],
    //
    mochaOpts: {
        ui: 'bdd',
        timeout: 60000
    },
}

```

**Describe the bug**
Not able to run Multiremote through wdio test runner. It throws error: 
`Error: Illegal key values seen in w3c capabilities: [browserA, browserB]`

**To Reproduce**
It is reproducible with official examples
1. Pull the wedriverio repo: https://github.com/webdriverio/webdriverio.git
2. `cd ./webdriverio`
3. `npm install`
4. `npm run setup`
5. Run the test: `npm run test:multiremote `

**Expected behavior**
2 browser have to be started

**Log**
Worker Error:
Error: Illegal key values seen in w3c capabilities: [browserA, browserB]
    at Request._callback /webdriverio/packages/webdriver/src/request.js:95:27)
    at Request.self.callback (/webdriverio/packages/webdriver/node_modules/request/request.js:185:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/webdriverio/packages/webdriver/node_modules/request/request.js:1161:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/webdriverio/packages/webdriver/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)

**Additional context**

> Please use the correct markdown

**Comment by christian-bromann** [comment]
You have set 
```js
    capabilities: {
        maxInstances: 2,
        myChromeBrowser: {
            desiredCapabilities: {
                browserName: 'chrome'
            }
        },
        myFirefoxBrowser: {
            desiredCapabilities: {
                browserName: 'firefox'
            }
        }
    },
```
as you can see you kept `maxInstances` in there which doesn't contain proper capabilities. With multiremote you should not set maxInstances. 

Join our  [![Gitter](https://badges.gitter.im/webdriverio/webdriverio.svg)](https://gitter.im/webdriverio/webdriverio?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) for these kind of questions. This is not necessary an issue.

**Comment by artyompvl** [comment]
@christian-bromann, thanks for the quick response but It is still reproducible without maxInstances. Also there is no `maxInstances ` in the examples https://github.com/webdriverio/webdriverio/blob/master/examples/wdio/multiremote/wdio.conf.js and it is reproducible there too. Can you please try to run the official example?
Steps:
1. Pull the wedriverio repo: `https://github.com/webdriverio/webdriverio.git`
2. cd ./webdriverio
3. npm install
4. npm run setup
5. Run the test: `npm run test:multiremote`

**Comment by goodbomb** [comment]
+1

@christian-bromann We're in the process of migrating to wdio v5, but this issue is a big blocker for us, as we have definite use cases for multi-remote. Migration is blocked until this issue can be fixed. Thanks!

**Comment by tlabeeuw** [comment]
@martomo and @christian-bromann Noticed that the attached #3288 changed the config structure slightly from:
```js
capabilities: {
  myBrowserA: {
    desiredCapabilities: {
      browserName: 'firefox'
    }
  },
  myBrowserB: {
    desiredCapabilities: {
      browserName: 'firefox'
    }
  }
}
```
to:
```js
capabilities: {
  myBrowserA: {
    capabilities: {
      browserName: 'firefox'
    }
  },
  myBrowserB: {
    capabilities: {
      browserName: 'firefox'
    }
  }
}
```
If your going to change it why not have it be more readable, and remove the double set of `capabilities`:
```js
capabilities: {
  myBrowserA: {
    browserName: 'firefox'
  },
  myBrowserB: {
    browserName: 'firefox'
  }
}
```

**Comment by tlabeeuw** [comment]
Thanks @christian-bromann for the quick reply! Most important is to get this out as it is indeed stopping our ability to upgrade to 5.x. Small side notes not necessary for us are, but to consider for the future:

* If the inner `capabilities` is necessary thats fine. Just looking at the documentation: https://webdriver.io/docs/multiremote.html#using-wdio-testrunner it seems inconsistent between the WDIO TestRunner config and calling `multiremote`, the former requires an inner `desiredCapabilities`, while the latter just `capabilities`. Think this is where my confusion stemmed from!
* The `resultChrome` is also inconsistent using the `browserName` field rather than then the name of the remote. I.e.: `myBrowserA.getText()` and `const result = browser.getText(); result.result.myBrowserA` mirror each other.
* Lastly will make a separate ticket for this, but the last line in that documentation states you cant run different remotes in parallel. Any reason we can't enable this? Would greatly speed up our already lengthy tests, and is more realistic to our apps parallel nature.

**Comment by christian-bromann** [comment]
> * the former requires an inner `desiredCapabilities`

This is not an option anymore and is a bug in the docs, mind making a PR?

> * The `resultChrome` is also inconsistent using the `browserName` field

Also feel free to suggest a change for it

> * Any reason we can't enable this?

You should be able to run remotes in parallel .. command calls on the `browser` object will trigger commands in all browser to run in parallel. Please note that you should use multiremote __only__ if you want to simulate a scenario where multiple users are required.

**Comment by tlabeeuw** [comment]
Thanks! Will submit a PR later for the doc changes, and a suggestion for the `result` PR. Described the parallel issue in greater detail in: #3322.

**Comment by martomo** [comment]
@tlabeeuw Thanks for pointing out the inconsistencies within the documentation, I've updated the multiremote configuration example for test runner also in #3288.

**Comment by christian-bromann** [comment]
This issue has been fixed within the last release.
